chaos_genius/core/utils/data_loader.py

```diff
@@ -80,24 +99,38 @@ def __init__(
     def _get_id_string(self, value):
         return f"{self.identifier}{value}{self.identifier}"
 
-    def _build_date_filter(self):
-        dt_col_str = self._get_id_string(self.dt_col)
+    def _convert_date_to_string(self, date: date, offset: str):
+        date = date.strftime("%Y-%m-%d")
+        date += f" 00:00:00 {offset}"
+        if not self.kpi_info.get("timezone_aware"):
+            date = (
+                pd.Timestamp(datetime.strptime(date, "%Y-%m-%d %H:%M:%S %z"))
+                .tz_convert(self.connection_info["database_timezone"])
+                .tz_localize(None)
+                .strftime("%Y-%m-%d %H:%M:%S")
+            )
```

Comment:
These formats look like ISO8601. Would be better to use `fromisoformat` and `isoformat` instead.

Also, there should be a better way to convert a date object to datetime with specific timezone without having to convert to string and back.

Reply:
ISO8061 follows a slightly different format.

While I agree with your point on not using strings for this task, we need to maintain backwards compatibility with the `SUPPORTED_TIMEZONES` variable for now which outputs an offset string. Once that is deprecated, we should switch to using datetime objects only. 

I'll add a TODO here.